cli_tools/gce_vm_image_import/importer/importer.go

```diff
@@ -137,14 +137,14 @@ func (i *importer) runProcess(ctx context.Context) error {
 func (i *importer) runStep(ctx context.Context, step func() error, cancel func(string) bool, getTraceLogs func() []string) (err error) {
 	e := make(chan error)
 	var wg sync.WaitGroup
-	wg.Add(1)
 	go func() {
 		//this select checks if context expired prior to runStep being called
 		//if not, step is run
 		select {
 		case <-ctx.Done():
 			e <- i.getCtxError(ctx)
```

Comment:
Is my understanding right? Here's my interpretation: 

1. Is the context cancelled? If so, return the context's cancellation error.
2. Otherwise, the step will run
3. Wait until either the step finishes, or the context is cancelled.

(This isn't feedback or a change request. Just seeing if my reading is right :)  )

Reply:
Yes, that's correct. The problem is WaitGroup got updated before it was determined if the context is already cancelled. This issue only occurs if import actually timed out. It doesn't affect imports that have enough time to finish.

Reply:
cool!